src/libraries/System.Private.CoreLib/src/System/MathF.cs

```diff
@@ -429,6 +429,18 @@ public static float Round(float x, int digits)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static float Round(float x, MidpointRounding mode)
         {
+            // Inline single-instruction modes
+            if (RuntimeHelpers.IsKnownConstant((int)mode))
```

Comment:
> This doesn't work with things that become constant after import, is that right?

No, it will work 🙂 If we were not able to fold `IsKnownConstant` to true in the importer, we convert it to GT_INTRINSIC and try again later during Global Morph.